core/templates/dev/head/App.ts

```diff
@@ -79,6 +79,9 @@ angular.module('oppia').config([
   function(
       $compileProvider, $cookiesProvider, $httpProvider,
       $interpolateProvider, $locationProvider, $provide) {
+    // Refer: https://docs.angularjs.org/guide/migration
+    // #migrate1.5to1.6-ng-services-$location.
+    $locationProvider.hashPrefix('');
```

Comment:
Can you explain why this exists and so does line 103-106?

Reply:
Added a comment.

103-106 was repeated, removed it now.